### DISCUSSION TITLE
How to implement a rate limiter/throttler?

### DISCUSSION THREAD

**Discussion by NikolaGrgic** [rwf2/Rocket#2014] [Questions]
Hey all,

I was working on a web API and wanted to know: how would I go about creating a guard to limit the rate at which a particular route can be called?

**Comment by Jackbaude** [comment]
I think the best way to do this is with your load balancer/ proxy (i.e nginx/ apache). Im not sure if Rocket can get the IP of a request coming through anyway.

**Reply by somehowchris** [reply]
I would go with something like nginx for the moment, yes hard to write custom hashing functions and it has no context of the api.

_Just a heads up, I‘m working on something to collect examples and best-practices like compression, cors and also rate limiting with rocket. But Since Im currently undergoing some exams that might take some months_

**Comment by najtin** [comment]
Maybe you could lock a Mutex at the start of the route function and drop it after a specified time in another thread.

**Reply by najtin** [reply]
ah that cant work, because one can not send a MutexGuard, sorry

**Reply by somehowchris** [reply]
Yea and this does not scale

**Comment by kolbma** [comment]
Like this...
https://crates.io/crates/rocket-governor

If you try it out, give a report how it works for you ;-)
